templates/flutter/lib/src/realtime_mixin.dart.twig

```diff
@@ -15,35 +15,46 @@ typedef GetFallbackCookie = String? Function();
 
 mixin RealtimeMixin {
   late Client client;
-  final Map<String, List<StreamController<RealtimeMessage>>> _channels = {};
+  final Set<String> _channels = {};
   WebSocketChannel? _websok;
   String? _lastUrl;
   late WebSocketFactory getWebSocket;
   GetFallbackCookie? getFallbackCookie;
   int? get closeCode => _websok?.closeCode;
+  int _subscriptionsCounter = 0;
+  Map<int, RealtimeSubscription> _subscriptions = {};
+  bool _notifyDone = true;
+  StreamSubscription? _websocketSubscription;
+  bool _creatingSocket = false;
 
   Future<dynamic> _closeConnection() async {
-    await _websok?.sink.close(normalClosure);
+    await _websocketSubscription?.cancel();
+    await _websok?.sink.close(status.normalClosure, 'Ending session');
     _lastUrl = null;
   }
 
   _createSocket() async {
+    if(_creatingSocket) return;
+    _creatingSocket = true;
```

Comment:
Now, when I [unsubscribe all](https://github.com/appwrite/playground-for-flutter/blob/8d330ce89e77501a0e63cd3729ebc575b5a14b0b/lib/main.dart#L242-L249), I get:

```
flutter: subscription: wss://cloud.appwrite.io/v1/realtime?project=delete
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: AppwriteException: , Missing channels (1008)
#0      RealtimeMixin.handleError (package:appwrite/src/realtime_mixin.dart:173:7)
#1      RealtimeMixin._createSocket.<anonymous closure> (package:appwrite/src/realtime_mixin.dart:61:13)
#2      _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
#3      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#4      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
#5      _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:123:11)
#6      _HandleErrorStream._handleData (dart:async/stream_pipe.dart:253:10)
#7      _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:153:13)
#8      _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
#9      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#10     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
#11     _Sync<…>
```

It's probably because the if here:

https://github.com/appwrite/sdk-generator/blob/b4b06a2b86e948005e9360b9344161c5c3e53f13/templates/flutter/lib/src/realtime_mixin.dart.twig#L151-L152

is ture for multiple iterations, but then when the `_createSocket()` is called, there are no more channels.

Reply:
Added check `_channels.isEmpty` in the `_createSocket` as well. That probably should resolve the issue.